build/dangerjs/detectNonApprovedDependencies/index.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  checkPackageVersionConstraints,
+  getVersionConstrains,
+  getRuntimeDependencies,
+  getPackageName,
+  FailedConstraintsExplanation,
+} from './utils'
+import config from '../../../config'
+
+const { paths } = config
+
+/**
+ * This check uses the following logic:
+ * - request runtime dependencies of @stardust-ui/react package (by crawling the code, starting from index file),
+ * - for each of the runtime dependencies:
+ *    - get corresponding set of version restrictions (by analyzing related package.json files),
+ *    - get list of approved dependency's versions,
+ *    - check if list of approved versions covers the set of version restrictions.
+ */
+const detectNonApprovedDependencies = async dangerJS => {
+  const { fail, markdown } = dangerJS
+  const failedVersionConstraints: FailedConstraintsExplanation[] = []
+
+  const dependencyPackageIds = getRuntimeDependencies('react')
```

Comment:
We use `dependencyPackageIds` only in this file and it is an array of strings:

```jsx
[
    'rtl-css-js@1.10.0',
    'scheduler@0.13.6',
    'webpack@4.35.0',
]
```

But in `checkPackageVersionConstraints` we do:
```jsx
const packageName = getPackageName(packageId)
```

`getPackageName()` is used only, so I think that we can change the structure to `object`:
```
{
  webpack: '4.35.0',
}
```

Reply:
I would like to leave it as it is for now, because this format resembles output of most related NPM utils, so it is quite easy to copy and paste their output to this file, without any additional parsing/composing/processing actions. This, essentially, was a key factor that made it very easy to me to prepare original version of this list.

However, if there will be strong need for providing strict data structure for this list, we could always rethink the approach.